spec/killbill_client/encoding_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'spec_helper'
+
+describe KillBillClient::API do
+
+  it 'should send double-encoded uri' do
+    plugin_properties = [
+        {
+            'key' => 'contractId',
+            'value' => "test"
+        },
+        {
+            'key'   => 'details',
+            'value' =>  {
+                'eventType' => 'voidEvent',
+                'transactionType' => 'void',
+                'contractType' => 'temp'
+            }.to_json
+        }
+    ].map{|hash| KVPair.from_hash(hash)}
```

Comment:
You should use `PluginPropertyAttributes` instead, for example:

* https://github.com/killbill/killbill-client-ruby/blob/05550b1eae4bd66affbbab548579ca541f91ea60/lib/killbill_client/models/payment_method.rb#L95-L99
* https://github.com/killbill/killbill-integration-tests/blob/1067b362ea5bf5a9c9b585f76fd450902d133341/killbill-integration-tests/plugins/payment/errors-handling/plugins_error_handling_base.rb#L58-L60